src/Scripting/ScriptingSystem.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <string_view>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <memory>
+#include <utility>
+#include <sol/sol.hpp>
+
+class IBindings;
+
+class ScriptingSystem {
+ public:
+    ScriptingSystem(
+        std::string_view scriptFolder,
+        const std::vector<std::string> &entryPointFiles
+    );
+
+    static std::unique_ptr<ScriptingSystem> create(
+        std::string_view scriptFolder,
+        const std::vector<std::string> &entryPointFiles);
+
+    void executeEntryPoints();
+
+    template<typename TBindings, typename ...TArgs>
+    void addBindings(std::string_view bindingTableName, TArgs &&... args) {
+        auto bindings = std::make_unique<TBindings>(_solState, std::forward<TArgs>(args) ...);
+        _addBindings(bindingTableName, std::move(bindings));
+    }
```

Comment:
I'm reading this an I have a question – is there a reason why bindings are lazy-initialized?

Reply:
I changed this part and now we're creating the binding table as soon as we call the Bindings instance. I still want to do a little improvement and use the lua `package` table to request the bindings from script.

With the improvement we'll have something like:
```lua
-- Instead of calling the function 
local bindings = requireGameBindings()
-- we could do
local bindings = require "bindings.game"
```
Which is going to be more natural when asking for module/libraries in the language. But I want to do that on another branch to keep it more atomic and more reviewable 😅